vortex-dtype/src/extension.rs

```diff
@@ -68,13 +68,14 @@ pub struct ExtDType {
 
 impl PartialEq for ExtDType {
     fn eq(&self, other: &Self) -> bool {
-        self.id == other.id
+        self.id == other.id && self.storage_dtype == other.storage_dtype
```

Comment:
at the very least, shouldn't we factor in the metadata? E.g. `vortex.timestamp` always stores i64, but the timezone is stored in the extmetadata. Should two timestamp types be equal if they have different timezones?